masonry/src/widget/widget.rs

```diff
@@ -165,20 +167,41 @@ pub trait Widget: AsAny {
 
     // --- Auto-generated implementations ---
 
-    // FIXME
-    #[cfg(FALSE)]
-    /// Return which child, if any, has the given `pos` in its layout rect.
+    /// Return which child, if any, has the given `pos` in its layout rect and clip path. In case
+    /// of overlapping children, the last child as determined by [`Widget::children_ids`] is
+    /// chosen.
+    ///
+    /// The child returned is a direct child, not e.g. a grand-child.
     ///
-    /// The child return is a direct child, not eg a grand-child. The position is in
-    /// relative coordinates. (Eg `(0,0)` is the top-left corner of `self`).
+    /// Has a default implementation that can be overridden to search children more efficiently.
```

Comment:
Should implementations overriding the default uphold certain conditions?

Custom implementations should probably uphold the guarantee that `pos` is actually inside the returned child's layout rect. Must custom custom implementations also uphold the guarantee that `pos` is contained in the clip path, and that in case of overlap the last child in "z-order" is chosen?

Reply:
Yes to all of these.

Ideally we should even implement some best-effort checks in debug_assertions mode to enforce that these conditions are upheld.

Reply:
The low effort version is to check the child contains the position in its layout rect and clip path.

The conditions uniquely identify a child, so we could do the default search alongside `Widget::get_child_at_pos` and check the same children are returned. For widgets with many children—where custom implementations are most likely—this would be slow. Valid options are to accept that cost or to limit this exhaustive check to widgets with N children for some value of N.

Reply:
Right, most of that is beyond the scope of this PR.

We could probably add a TODO comment and keep it to that for now.

> Valid options are to accept that cost or to limit this exhaustive check to widgets with N children for some value of N.

Sure. I was thinking we'd probably need to add a `child_count()` method for cases where building a child array is needlessly expensive (eg a Flex with thousands of children).

Reply:
Added a todo comment, so we can leave this for a future PR